sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -145,14 +145,14 @@ trait ProgressReporter extends Logging {
   /** Finalizes the query progress and adds it to list of recent status updates. */
   protected def finishTrigger(hasNewData: Boolean): Unit = {
     assert(currentTriggerStartOffsets != null && currentTriggerEndOffsets != null)
-    currentTriggerEndTimestamp = triggerClock.getTimeMillis()
+    currentTriggerEndTimestamp = triggerClock.nanoTime()
 
     val executionStats = extractExecutionStats(hasNewData)
     val processingTimeSec =
-      (currentTriggerEndTimestamp - currentTriggerStartTimestamp).toDouble / MILLIS_PER_SECOND
+      (currentTriggerEndTimestamp - currentTriggerStartTimestamp).toDouble / NANOS_PER_SECOND
 
     val inputTimeSec = if (lastTriggerStartTimestamp >= 0) {
```

Comment:
When we change to nanoTime() it's going to be no longer correct. (negative value is also a valid for `nanoTime()`) Change it to `Option[Long]`.